### PR TITLE
[MAISTRA-840] Remove the login strategy from Kiali auth CR

### PR DESCRIPTION
**Pull request by jwendell**
The only supported authentication strategy is now "openshift".
This aligns with what we have for other components already, like
Prometheus, Grafana and Jaeger.

### CODE REVIEW

File: deploy/examples/maistra_v1_servicemeshcontrolplane_cr_auth.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ spec:
   # element too.  An empty element is interpreted as null and will override all
   # default values (i.e. no values will be specified for that element, not even
   # the defaults baked into the chart values.yaml).
+
+  # For a complete list of all options, see the docs at https://maistra.io/docs/getting_started/custom-install/
```
Comment (brian-avery) [inline deploy/examples/maistra_v1_servicemeshcontrolplane_cr_auth.yaml:11]: Can you please update docs as well?

Comment (brian-avery) [reply deploy/examples/maistra_v1_servicemeshcontrolplane_cr_auth.yaml:11]: Other than that, lgtm

Comment (jwendell) [reply deploy/examples/maistra_v1_servicemeshcontrolplane_cr_auth.yaml:11]: It will be part of maistra-787

**Review by brian-avery — APPROVED** [review]
Content lgtm. This should be merged alongside the PR for maistra-787

File: helm/istio/charts/kiali/templates/kiali-cr.yaml [hunk 2]
```diff
@@ -8,14 +8,10 @@ spec:
   istio_namespace: "{{ .Release.Namespace }}"
 
   auth:
-{{- if .Values.dashboard }}
```
Comment (rcernich) [inline helm/istio/charts/kiali/templates/kiali-cr.yaml:11]: The only change here should be adding `.Values.dashboard.secretName` to the `and` statement below.  If not all of those are set, we should be using oauth (`openshift`) by default.

**Review by rcernich — CHANGES_REQUESTED** [review]

**Review by rcernich — APPROVED** [review]

**Review by knrc — APPROVED** [review]

**Pull request merged**